### DISCUSSION TITLE
Create a common library/catalog of references (NIST standards; EAC guidelines; etc.)

### DISCUSSION THREAD

**Discussion by cwulfman** [ion-oset/developer-commons#14] [Design & Development]
@stratofax  has created a [Government Standards page on the wiki ](https://github.com/ion-oset/developer-commons/wiki/Government-Standards), which heavily references [his personal repo of election test cases](https://github.com/ion-oset/election-test-cases), which was created as part of a sprint several months ago. This repo overlaps the more recent [NIST-1500-100-103-examples](https://github.com/TrustTheVote-Project/NIST-1500-100-103-examples) repository, which is being referenced by the front-end developers.

I propose that these two resources be merged into a new TTV-managed repository, one that has a clear purpose and can be maintained and updated by TTV.

**Comment by stratofax** [comment]
Just to clarify, I actually created the [Government Standards page on the wiki ](https://github.com/ion-oset/developer-commons/wiki/Government-Standards), based on links that @ion-oset sent over in Slack. Any redundancy is therefore a result of my ignorance. I do agree that the sooner we move the developer-commons under the TTV flag, the better -- perhaps the [NIST-1500-100-103-examples](https://github.com/TrustTheVote-Project/NIST-1500-100-103-examples) repository can be incorporated into the "official" TTV developer commons repo?

**Reply by cwulfman** [reply]
No bad! I agree that the NIST-1500-100-103-examples repo should be reconstituted in the commons repo.

**Reply by stratofax** [reply]
It seems like it's just a matter of getting John Sebes to approve the transfer of developer-commons to the TTV account.

**Comment by cwulfman** [comment]
Now that we're gathering the various component repositories under the same roof, I think we can revive this discussion, which I have renamed to reflect its broader scope.

I propose we develop an organization-wide repository of commonly used documentation: or, better still, a catalog of documentation with links to canonical versions where they exist, and to a single local copy where they do not. Just to get the ball rolling, let me throw out a concrete proposal:

Create a repo in the TrustTheVote Project GitHub organization called "Standards and References". Make the README for this repo a catalog (or annotated bibliography, or what you will) of documentation relevant to ElectOS development. This is not a list of programming best practices or how-tos; we're already building such a developer-centric resource in Developer Commons.

Catalog entries should link to the canonical (published) version of a reference whenever possible.  When it seems appropriate (e.g., because the source is not stable), put a copy of the reference in the repo in a "library" directory.

**Reply by ion-oset** [reply]
I _definitely_ a 👍🏼 on the catalog. It lets us track which documents are authoritative with the addition of our own notes and gives us a level of indirection that preserves what we know as docs are updated or move around and as tools change. Ideally it's in some format that is machine readable, either one that can used to generate a human readable one, or something that is effectively both. I'd vote for whatever bibliographical format that makes sense, especially if it can be transformed into Markdown, HTML and/or PDF. I defer to your expertise on the choice on the format, though I would be in favor of tools that are common and open, and can be invoked programmatically or from automation.

I can also see some value to having local copies so we don't have anything critical disappear without a backup. We would need to manage the usual problem of out-of-date local copies being taken as authoritative. (We've already run into this. as you pointed out, with the various versions of NIST SP-1500-100 we are referencing.) We can probably manage that by having the catalog distinguish local from official copies, and explicitly highlighting or marking the authoritative form.

I _was_ thinking there was a case for using the Developer Commons repository to store these documents, but you are probably right that Standards and References should be its own place. Among other things it solves the issue of having large binary documents mixed in a repository that people are likely to clone often.
